flang/lib/Lower/OpenMP.cpp

```diff
@@ -716,6 +727,67 @@ genOMP(Fortran::lower::AbstractConverter &converter,
   createBodyOfOp<omp::CriticalOp>(criticalOp, converter, currentLocation, eval);
 }
 
+ void Fortran::lower::genOpenMPSectionsBlock(Fortran::lower::AbstractConverter &converter, Fortran::lower::pft::Evaluation &eval){
+     auto &firOpBuilder = converter.getFirOpBuilder(); 
+     auto currentLocation = converter.getCurrentLocation();
+     mlir::omp::SectionOp sectionOp = firOpBuilder.create<mlir::omp::SectionOp>(currentLocation);
+     createBodyOfOp<omp::SectionOp>(sectionOp, converter, currentLocation, eval);
+ }
+
+ // TODO: Add support for reduction
+ static void
+ genOMP(Fortran::lower::AbstractConverter &converter,
+        Fortran::lower::pft::Evaluation &eval,
+        const Fortran::parser::OpenMPSectionsConstruct &sectionsConstruct) {
+     auto &firOpBuilder = converter.getFirOpBuilder();
+     auto currentLocation = converter.getCurrentLocation();
+     SmallVector<Value> privateClauseOperands, firstPrivateClauseOperands,
+         lastPrivateClauseOperands, reductionVars, allocateOperands,
+         allocatorOperands;
+     mlir::Attribute noWaitClauseOperand;
+     const auto &sectionsClauseList = std::get<Fortran::parser::OmpClauseList>(
+             std::get<Fortran::parser::OmpBeginSectionsDirective>(sectionsConstruct.t).t);
+
+     for(const auto &clause : sectionsClauseList.v){
+
+       if (std::get_if<Fortran::parser::OmpClause::Lastprivate>(&clause.u)) {
+         TODO(currentLocation, "OMPC_LastPrivate");
```

Comment:
Also should these TODOs be added in case of loop construct? Example: https://github.com/flang-compiler/f18-llvm-project/blob/0ee6c0f4a2c05f35e27d3014b9caa8d4ecd12fd0/flang/lib/Lower/OpenMP.cpp#L515